packages/medusa/src/api-v2/admin/fulfillments/[id]/route.ts

```diff
@@ -0,0 +1,38 @@
+import { cancelFulfillmentsWorkflow } from "@medusajs/core-flows"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  AdminFulfillmentDeleteResponse,
+  IFulfillmentModuleService,
+} from "@medusajs/types"
+
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../types/routing"
+
+export const DELETE = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse<AdminFulfillmentDeleteResponse>
+) => {
+  const { id } = req.params
+  const fulfillmentModuleService = req.scope.resolve<IFulfillmentModuleService>(
+    ModuleRegistrationName.FULFILLMENT
+  )
+
+  await fulfillmentModuleService.retrieveFulfillment(id)
```

Comment:
Is this only for validation? If so, I'd suggest we do it in the workflow instead

Reply:
I have come across a similar case a few times, and I can't figure out what the right thing to do is, so maybe we should establish a convention.

Typically, our workflows expect a specific shape of data for bulk mutating a resource and therefore do not have validation built-in – as they would have in the case of a single mutation through `retrieve`. Given that our HTTP almost always operates on a single (root) resource, one could argue the validation is well-placed in the controller.

E.g. sending a request to update a product with an ID that does not exist will perform a selector-based update. There is not a product to update, so the return value of the workflow in an empty array and the subsequent call to fetch the product fails.

Reply:
@olivermrbl you have a point due to the fact that we do a `select`-based mutation. My thought-process is that workflows should validate that what they are operating on is correct, since they can be executed from a non-http flow, but in this specific scenario I can see why it might make sense to do the validation here, cc @riqwan WDYT?

Reply:
I think its a moot point for this endpoint since the module internally calls `retrieveFulfillment` to validate its existence.

I agree that the validation generally being on the workflow makes total sense as its built to run independently, but running validations on the endpoint is also totally valid in cases where the endpoint needs to have agency over its job of rendering the response graciously and consistently.

Reply:
The only aspect to consider here is to not fire 3-4 DB requests for just checking if something exists, which is why I suggested moving the check elsewhere. Ideally the HTTP endpoint can "clean up" the response in a purely synchronous manner, but I agree that there might be times where an async call to some endpoint is needed for a better response.

Anyhow, it's not too important for now, but these are considerations we'll need to make as we make v2.0 more prod-ready